lib/temp-directory.js

```diff
@@ -9,12 +9,18 @@ var osenv = require('osenv');
 
 function create(context, next) {
   var _path = path.join(osenv.tmpdir(),uuid.v4());
-  var npmConfigTmp = path.join(_path, 'npm_config_tmp');
+  var npmConfigTmp = '';
 
-  context.path = _path;
-  context.npmConfigTmp = npmConfigTmp;
+  if (context.options && context.options.tmpDir) {
+    context.path = path.join(context.options.tmpDir, context.module.name);
+    npmConfigTmp = path.join(context.path, 'npm_config_tmp');
+  } else {
+    npmConfigTmp = path.join(_path, 'npm_config_tmp');
+    context.path = _path;
+  }
```

Comment:
Also, with a custom path, we should likely still be using a UUID of some sort to make the folder unique